etna/models/base.py

```diff
@@ -619,7 +703,7 @@ def validation_step(self, batch: dict, *args, **kwargs):  # type: ignore
         return loss
 
 
-class DeepBaseModel(FitAbstractModel, DeepBaseAbstractModel, BaseMixin):
+class DeepBaseModel(DeepBaseAbstractModel, NonPredictionIntervalContextRequiredAbstractModel):
```

Comment:
Because we define `forecast` already in this class. The signature is already fixed. If we want to add intervals we should rewrite this class. It is probably isn't matter of this task.